NitroxClient/Helpers/NitroxProtobufSerializer.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using ProtoBuf;
+using ProtoBuf.Meta;
+
+namespace NitroxClient.Helpers
+{
+    public class NitroxProtobufSerializer
+    {
+        public readonly RuntimeTypeModel model;
+        public readonly Dictionary<Type, int> NitroxTypes = new Dictionary<Type, int>();
+
+        private readonly Dictionary<Type, int> knownTypes;
+
+        public static NitroxProtobufSerializer Main;
+
+        protected RuntimeTypeModel Model { get { return model; } }
+
+        public NitroxProtobufSerializer(params string[] assemblies)
+        {
+            Main = this;
+            model = TypeModel.Create();
+            knownTypes = (Dictionary<Type, int>)typeof(ProtobufSerializerPrecompiled).GetField("knownTypes", BindingFlags.NonPublic | BindingFlags.Static).GetValue(null);
+
+            foreach (string assembly in assemblies)
+            {
+                RegisterAssemblyClasses(assembly);
+            }
+
+            foreach (Type type in Assembly.GetExecutingAssembly().GetTypes())
+            {
+                bool hasUweProtobuf = (type.GetCustomAttributes(typeof(ProtoContractAttribute), true).Length > 0);
+
+                if (hasUweProtobuf)
+                {
+                    // As of the latest protobuf update they will automatically register detected attributes.
+                    model.Add(type, true);
+                    knownTypes[type] = int.MaxValue; // UWE precompiled is going to pass everything to us
+                    NitroxTypes[type] = int.MaxValue;
```

Comment:
minor: extract common code into a method

Reply:
Moved to AddType method